examples/step-63/step-63.cc

```diff
@@ -769,6 +846,35 @@ namespace Step63
   }
 
 
+  // @sect4{<code>AdvectionProblem::setup_smoother</code>}
+
+  // Here we setup the smoother based on the settings in the .prm. The two
+  // options that are of significance is the number of pre- and post-smoothing
+  // steps on each level of the multigrid v-cycle and the relaxation parameter.
+
+  // Since multiplicative methods tend to be more powerful than additive method,
+  // fewer smoothing steps are required to see convergence indepedent of mesh
+  // size. The same hold for block smoothers over point smoothers. This is
+  // reflected in the choice for the number of smoothing steps for each type of
+  // smoother below.
+
+  // The relaxation parameter for point smoothers is chosen based on trial and
+  // error, and they reflect values necessary to keep the iteration counts in
+  // the GMRES solve constant (or as close as possible) as we refine the mesh.
+  // The two values given for both ``Jacobi" and ``SOR" are for degree 1 and
+  // degree 3 finite elements. If the user wants to change to another degree,
+  // they may need to adjust these numbers. For block smoothers, this parameter
+  // has a more straightforward interpretation, namely that for additive methods
+  // in 2D, a DoF can have a repeated contribution from up to 4 cells,
+  // therefore we must relax these methods by 0.25 to compensate. This is not an
+  // issue for multiplicative methods as each cell inverse application carries
+  // new information to all its DoFs.
+
+  // Finally, as mention above, the point smoothers only operate on DoFs, and
+  // the block smoothers on cells, so only the block smoothers need to be given
+  // information regarding cell orderings. DoF ordering for point smoothers has
+  // already been taken care of in setup_system().
+
```

Comment:
Here I basically say "we get these param values by trial and error". Did I say enough/too much? Anything important I'm leaving out?